addons/loyalty/models/base_partner_merge.py

```diff
@@ -0,0 +1,45 @@
+from odoo import models
+
+
+class MergePartnerAutomatic(models.TransientModel):
+    _inherit = 'base.partner.merge.automatic.wizard'
+
+    def _update_foreign_keys(self, src_partners, dst_partner):
+        """ Override of base to merge corresponding nominative loyalty cards."""
+        self._merge_loyalty_cards(src_partners, dst_partner)
+        super()._update_foreign_keys(src_partners, dst_partner)
+
+    def _merge_loyalty_cards(self, src_partners, dst_partner):
+        """ Merge nominative loyalty cards.
+
+        :param src_partners: recordset of source res.partner records to merge
+        :param dst_partner: destination res.partner record
+        """
+        LoyaltyCard = self.env['loyalty.card']
+        cards_per_program = dict(
+            LoyaltyCard._read_group(
+                domain=[
+                    ('partner_id', 'in', src_partners.ids),
+                    '|',
+                        ('program_id.applies_on', '=', 'both'),
+                        '&',
+                            ('program_id.program_type', 'in', ('ewallet', 'loyalty')),
+                            ('program_id.applies_on', '=', 'future'),
+                ],
+                groupby=['program_id'],
+                aggregates=['id:recordset'],
+            )
+        )
+        for program, cards in cards_per_program.items():
+            total_points = sum(card.points for card in cards)
+            dst_card = LoyaltyCard.search([
+                ('partner_id', '=', dst_partner.id),
+                ('program_id', '=', program.id),
+            ], limit=1)
```

Comment:
As `dst_cart` is already included in `cards`, why can't you find it in the `cards` variable instead of making another query here?  It would be better if we can avoid making queries in the loop here.

Reply:
The `cards` variable does not include `dst_card` because it is in the `cards_per_program` dictionary, which is the output of the `_read_group` method. This method specifically groups the cards of only `src_partners`.

Reply:
or we can get `dst_card` in the sane `_read_group` call by adding `('partner_id', '=', dst_partner.id)` in the domain ?

Reply:
Indeed, I was confused by the `(cards - final_card)` below.
As we probably don't merge that many partners at once, let's keep it like this for now then, we'll improve it in the future if customers complain about the performance :ok_hand:

Reply:
Yes, I have verified that currently we allow a maximum of 3 partners (including source and destination) to merge.